ui/lesson/ScriptingChallenge/index.tsx

```diff
@@ -79,6 +80,18 @@ export default function ScriptingChallenge({
   const [challengeSuccess, setChallengeSuccess] = useState(false)
   const [hydrated, setHydrated] = useState(false)
   const [errors, setErrors] = useState<string[]>([])
+  const debouncedCode = useDebounce(code, 500)
```

Comment:
Debouncing so that we aren't constantly writing to local storage on every keypress